content/docs/practices/alerting.md

```diff
@@ -0,0 +1,71 @@
+---
+title: Alerting
+sort_rank: 4
+---
+
+# Alerting
+
+We recommend that you read [My Philosophy on Alerting](https://docs.google.com/a/boxever.com/document/d/199PqyG3UsyXlwieHaqbGiWVa8eMWi8zzAn0YfcApr8Q/edit).
+
+To summarize, keep your alerting simple, alert on symptoms, have good consoles
+to allow you to pinpoint causes and avoid having pages where there's nothing
+for you to do.
+
+## What to Alert On
+
+Aim to have as few alerts as you can, by alerting on symptoms that are
+associated with end-user pain rather than trying to catch every possible way
+that pain could be caused. Your consoles should be linked form your alerts,
+and make it easy to figure out which component is at fault.
+
+Always allow slack in your alerting for small blips.
+
+### Online Serving Systems
+
+Typically you want to alert on high latency and error rates as high up in the stack as possible.
+
+You should only page on latency at one point in your stack, if a lower
+component is slower than it should be but the overall user latency is fine then
+there's no need to page.
+
+For error rates, you should page on errors to the user. If there's errors
+further down the stack that'll cause such a failure, there's no need to page on
+them seperately. However if some failures don't cause a failure to the user but
+are otherwise bed enough to require human involvment, you'll need pages to be
+sent on those.
+
+You may need alerts for different types of request if they've different
+characteristics, or problems in a low-traffic type of request would be drowned
+out by high-traffic requests.
+
+### Offline Processing
+
+The key metric is how long data takes to get through the system, so you should
+page if that gets high enough to cause user impact.
+
+### Batch Jobs
+
+You should page if the batch job hasn't suceeded recently enough, and this will
+cause problems for the user.
+
+This should generally be at least enough time for 2 full runs of the batch job. For a job that runs every 4 hours and takes an hour, 10 hours would be a reasonable threshold.
+If you can't withstand a single run failing, you should run the job more often as a single failure shouldn't require human interventino.
+
+### Capacity
+
+While not an problem causing immediate user impact, being close to capacity often requires human intervention to avoid an outage in the near future.
+
+### Metamonitoring
+
+It's important to have confidence that your monitoring is working, accordingly
+you should have alerts that there's sufficient Prometheuses, Alertmanagers,
+PushGateways and other monitoring infrastructure running to alert on any
+problems.
+
+As always, if you can alert on symptoms rather than causes this helps reduce
+noise. For example a blackbox test that alerts are getting from PushGateway to
```

Comment:
We're referring to our particular Alertmanager and PushGateway implementations, thus I think they're proper nouns and get capitals.